File: 141.cycled_linked_list.cpp
class Solution
{
public:
    bool hasCycle(ListNode *head)
    {
        ListNode *slow = head;
        ListNode *fast = head;
        while (fast != NULL && slow != NULL)
        {
            slow = slow->next;
            fast = fast->next->next;

            if (fast == slow)
            {
                return true;
            }
        }
        return false;
    }
}

File: 141.cycled_linked_list.js
const hasCycle =(head)=>{
    let slow = head;
    let fast = head;
    while (fast !== null && slow !== null) {
        slow = slow.next;
        fast = fast?.next?.next;

        if (fast === slow) {
            return true;
        }
    }
    return false;
}
